src/utils.js

```diff
@@ -124,25 +124,24 @@ export const normalizeData = function(data, baseUrl) {
       data._components = {};
       for (const [key_, value_] of Object.entries(value)) {
         if (value_ !== null) {
-          data._components[key_] = normalizeData(value_, baseUrl);
+          data._components[key_] = normalizeData(value_, baseUrl, depth + 1);
           data._components[key_]._path = data._path;
         }
       }
       delete data[key];
     } else if (key === 'items' && value) {
-      data[key] = value.map(item => normalizeData(item, baseUrl));
+      data[key] = value.map(item => normalizeData(item, baseUrl, depth + 1));
     } else if (new Set(['id', 'parent', 'children']).has(key)) {
       if (key === 'parent') {
-        data[`_${key}`] = normalizeData(value, baseUrl);
+        data[`_${key}`] = normalizeData(value, baseUrl, depth + 1);
       } else {
         data[`_${key}`] = value;
       }
       delete data[key];
     } else if (key.length && key[0] === '@') {
-      if (key === 'id') {
-        data.id = value;
-      } else {
```

Comment:
These lines had no effect so they were safely removed. (node.id is set explictly in node creation functions)